source/raylib/raylib_types.d

```diff
@@ -63,6 +63,54 @@ struct Rectangle
     float height;
     alias w = width;
     alias h = height;
+
+    Vector2 origin() { // Rectangle function exclusive to raylib-d
+        return Vector2(x, y);
+    }
+    
+    Vector2 dimensions() {
+        return Vector2(width, height);
+    }
+
+    alias topLeft = origin;
+
+    Vector2 topRight() const {
+        return Vector2(x:(x + width), y:y);
+    }
+
+    Vector2 bottomLeft() const {
+        return Vector2(x:x, y:(y + height));
+    }
+
+    Vector2 bottomRight() const {
+        return Vector2(x:(x + width), y:(y + height));
+    }
+
+    void opOpAssign(string op)(Vector2 offset) {
+        static assert(op=="+" || op=="-");
+        
+        static if (op=="+") {
+            this.x += offset.x;
+            this.y += offset.y;
+        } else static if (op=="-") {
+            this.x -= offset.x;
+            this.y -= offset.y;
+        }
+    }
+
+    Rectangle opBinary(string op)(Vector2 offset) const {
+        static assert(op=="+" || op=="-");
+
+        Rectangle result = this;
+        static if (op=="+") {
+            result.x += offset.x;
+            result.y += offset.y;
+        } else static if (op=="-") {
+            result.x -= offset.x;
+            result.y -= offset.y;
+        }
```

Comment:
```suggestion
        result += offset;
```

Reply:
Wouldn't this change result in `+=` being equivalent to `-=`?

Reply:
Oh right, I forgot it has to be a mixin.

```d
mixin("result ", op, "= offset;");
```

Reply:
Or even
```d
result.opOpAssign!(op)(offset);
```